services/spar/src/Spar/Scim/User.hs

```diff
@@ -315,24 +283,24 @@ updateValidScimUser tokinfo uidText newScimUser = do
     -- TODO: how do we get this safe w.r.t. race conditions / crashes?
 
     -- construct old and new user values with metadata.
-    uid :: UserId <- parseUid uidText
-    oldScimStoredUser :: Scim.StoredUser ScimUserExtra
-      <- let err = throwError $ Scim.notFound "user" uidText
-         in maybe err pure =<< Scim.get tokinfo uidText
+    oldScimStoredUser :: Scim.StoredUser SparTag
+      <- Scim.getUser tokinfo uid
 
-    let userRef = newScimUser ^. vsuSAMLUserRef
-    assertUserRefUnused uid userRef
+    assertUserRefMatchesUserId (newScimUser ^. vsuSAMLUserRef) uid
 
     if Scim.value (Scim.thing oldScimStoredUser) == (newScimUser ^. vsuUser)
       then pure oldScimStoredUser
       else do
-        newScimStoredUser :: Scim.StoredUser ScimUserExtra
+        newScimStoredUser :: Scim.StoredUser SparTag
           <- lift $ updScimStoredUser (newScimUser ^. vsuUser) oldScimStoredUser
 
         -- update 'SAML.UserRef'
-        let uref = newScimUser ^. vsuSAMLUserRef
-        lift . wrapMonadClient $ Data.insertSAMLUser uref uid  -- on spar
-        bindok <- lift $ Intra.Brig.bindBrigUser uid uref  -- on brig
+        let newuref = newScimUser ^. vsuSAMLUserRef
+        lift . wrapMonadClient $ Data.insertSAMLUser newuref uid  -- on spar
+        -- TODO: compute UserRef from oldScimStoredUser and call Data.deleteSAMLUser on that
+        -- (exercise to the reader: what goes wrong if we don't?  it has nothin to do with the
```

Comment:
~Life~ Reading code is hard enough as it is, let's not have exercises for the reader.